addons/account/models/account_move.py

```diff
@@ -894,11 +894,55 @@ def _reconcile_lines(self, debit_moves, credit_moves, field):
             credit_move = credit_moves[0]
             company_currency = debit_move.company_id.currency_id
             # We need those temporary value otherwise the computation might be wrong below
-            temp_amount_residual = min(debit_move.amount_residual, -credit_move.amount_residual)
             temp_amount_residual_currency = min(debit_move.amount_residual_currency, -credit_move.amount_residual_currency)
             dc_vals[(debit_move.id, credit_move.id)] = (debit_move, credit_move, temp_amount_residual_currency)
             amount_reconcile = min(debit_move[field], -credit_move[field])
```

Comment:
This assignment should be done inside those new conditions below.

Reply:
If we are to compare apples to apples and orange to oranges, if the debit is a EUR and the credit is AUD we cannot fetch the min() here.